src/nvim/plines.c

```diff
@@ -51,7 +51,8 @@ int plines_win(win_T *wp, linenr_T lnum, bool winheight)
 /// @return Number of filler lines above lnum
 int win_get_fill(win_T *wp, linenr_T lnum)
 {
-  int virt_lines = decor_virt_lines(wp, lnum, NULL);
+  foldinfo_T fi = fold_info(wp, lnum);
```

Comment:
`win_get_fill()` is used quite often. Will using `fold_info()` here cause performance problems?

Reply:
`hasFoldingWin()` with `cache` is used for diff:
https://github.com/neovim/neovim/blob/0f52e2c849cbd923fc30b8bbef088b6f7642aeab/src/nvim/diff.c#L2137-L2140

Reply:
The cache seems to work fine for the added tests, I can't judge right now if it will always be correct.